pkg/eth/api.go

```diff
@@ -524,12 +524,18 @@ func (pea *PublicEthAPI) localGetTransactionReceipt(ctx context.Context, hash co
 		"logsBloom":         receipt.Bloom,
 	}
 
-	// Assign receipt status or post state.
-	if len(receipt.PostState) > 0 {
+	if blockNumber <= pea.B.Config.ChainConfig.ByzantiumBlock.Uint64() {
```

Comment:
A problem with this is that this deviates from the normal, expected behavior of this eth json-rpc endpoint. The regular endpoint only fills in the status if it is explicitly known: https://github.com/ethereum/go-ethereum/blob/master/internal/ethapi/api.go#L1650. So this breaks compatibility with the regular endpoint.